mozilla_django_oidc_db/plugins.py

```diff
@@ -49,48 +45,71 @@ def validate_settings(self) -> None:
         ...
 
     @abstractmethod
-    def filter_users_by_claims(self, claims: JSONObject) -> UserManager[AbstractUser]:
-        """Return all users matching the specified subject."""
-        ...
+    def handle_callback(self, request: HttpRequest) -> HttpResponse:
+        """Return an HttpResponse based on the callback view specified on the plugin.
 
-    @abstractmethod
-    def create_user(self, claims: JSONObject) -> AbstractUser | AnonymousUser:
-        """Create an authenticated user.
+        .. code:: python
 
-        This returns either a User if we need to have a new user created.
-        Otherwise, for application that don't need to create a new user whenever there is a login,
-        an AnonymousUser is returned.
-        """
-        ...
+           def view(self, request: HttpRequest) -> HttpResponse:
+               return admin_callback_view(request)
 
-    @abstractmethod
-    def update_user(self, user: AbstractUser, claims: JSONObject) -> AbstractUser:
-        """
-        Update existing user with new claims, if necessary save, and return user.
         """
         ...
 
     @abstractmethod
-    def handle_callback(self, request: HttpRequest) -> HttpResponse:
-        """Return an HttpResponse based on the callback view specified on the plugin.
+    def get_extra_params(self, request: HttpRequest, extra_params: dict) -> dict: ...
 
-        For example, for class based views:
 
-        .. code:: python
+class BaseOIDCPlugin:
+    def __init__(self, identifier: str):
+        self.identifier = identifier
 
-           def view(self, request: HttpRequest) -> HttpResponse:
-               view = self.callback_view.as_view()
-               return view(request)
+    def get_config(self) -> OIDCClient:
+        return OIDCClient.objects.resolve(self.identifier)
 
-        """
+    def get_setting(self, attr: str, *args) -> Any:
+        config = self.get_config()
+
+        return get_setting_from_config(config, attr, *args)
+
+    def get_extra_params(self, request: HttpRequest, extra_params: dict) -> dict:
+        return extra_params
+
+
+@runtime_checkable
+class AnonymousUserOIDCPluginProtocol(OIDCBasePluginProtocol, Protocol):
+    def get_or_create_user(
+        self,
+        access_token: str,
+        id_token: str,
+        payload: JSONObject,
+        request: HttpRequest,
+    ) -> AnonymousUser | None: ...
+
+
+@runtime_checkable
+class AbstractUserOIDCPluginProtocol(OIDCBasePluginProtocol, Protocol):
+    def create_user(self, claims: JSONObject) -> AbstractUser: ...
+
+    def update_user(self, user: AbstractUser, claims: JSONObject) -> AbstractUser: ...
+
+    def filter_users_by_claims(self, claims: JSONObject) -> UserManager[AbstractUser]:
```

Comment:
I'm not sure if that `UserManager` will work because the manager is typically a custom one in all our projects: https://bitbucket.org/maykinmedia/default-project/src/b33f82cf9939081717710c249ae15ff6af561f31/src/project_name/accounts/managers.py#lines-4

perhaps `models.Manager[AbstractUser]` is more robust?